src/logstatsexp.jl

```diff
@@ -0,0 +1,38 @@
+"""
+    logmeanexp(A; dims=:)
+
+Computes `log.(mean(exp.(A); dims))`, in a numerically stable way.
+"""
+function logmeanexp(A::AbstractArray; dims=:)
+    R = logsumexp(A; dims)
+    N = length(A) ÷ length(R)
+    return R .- log(N)
+end
+
+"""
+    logvarexp(A; dims=:)
+
+Computes `log.(var(exp.(A); dims))`, in a numerically stable way.
+"""
+function logvarexp(
+    A::AbstractArray; dims=:, corrected::Bool=true, logmean=logmeanexp(A; dims)
```

Comment:
We make sure to use a special single pass algorithm for logsumexp for performance reasons, so it seems a bit unfortunate to iterate over the data multiple times and precompute logmean by default. Could we use a single pass algorithm for the variance as well?